server/channels/store/sqlstore/team_store.go

```diff
@@ -1513,6 +1513,46 @@ func (s SqlTeamStore) GetCommonTeamIDsForTwoUsers(userID, otherUserID string) ([
 	return teamIDs, nil
 }
 
+func (s SqlTeamStore) GetCommonTeamIDsForMultipleUsers(userIDs []string) ([]string, error) {
```

Comment:
In my testing, it wasn't a heavy query, even when there are hundreds of team memberships involved. And since this is not called often, so we can skip the cache for now I believe.